sycl/source/detail/get_backend.hpp

```diff
@@ -0,0 +1,29 @@
+//==------------------ get_backend.hpp - get impls backend
```

Comment:
Maybe call this just "detail/backend.hpp" since there might be other functionality wanted here in addition to "get"

Reply:
Are the same names of files in include and source good?

Reply:
I'd say better call it "backend_impl.hpp"